webpack/assets/javascripts/react_app/components/AutoCompleteSearch/AutoCompleteReducer.js

```diff
@@ -0,0 +1,59 @@
+import Immutable from 'seamless-immutable';
+import { STATUS } from '../../constants';
+import {
+  AUTO_COMPLETE_REQUEST,
+  AUTO_COMPLETE_SUCCESS,
+  AUTO_COMPLETE_FAILURE,
+  AUTO_COMPLETE_RESET,
+} from './AutoCompleteConstants';
+
+const initialState = Immutable({
+  controller: '',
+  error: null,
+  results: [],
+  searchQuery: null,
+  status: STATUS.RESOLVED,
+  trigger: null,
+});
+
+export default (state = initialState, action) => {
+  const { payload, type } = action;
+  if (!payload) {
+    return state;
```

Comment:
the state returns in the default switch case, is it really needed to return it here as well?